### PR TITLE
Fix query interface

### PR DESCRIPTION
**Pull request by douglasselias**

### CODE REVIEW

**Comment by douglasselias** [comment]
Both tests for query interface [v5](https://github.com/cockroachdb/sequelize-cockroachdb/pull/60/checks?check_run_id=1973326480) and [v6](https://github.com/cockroachdb/sequelize-cockroachdb/pull/60/checks?check_run_id=1973328921) are not fully working because the patch is in this [PR](https://github.com/cockroachdb/sequelize-cockroachdb/pull/56)

File: tests/query_interface_test.js [hunk 1]
```diff
@@ -0,0 +1,587 @@
...
+          throw new Error('Error not thrown...');
+        } catch (error) {
+          expect(error).to.be.instanceOf(Sequelize.UnknownConstraintError);
+          expect(error.table).to.equal('users');
+          expect(error.constraint).to.equal('unknown__constraint__name');
+        }
+      });
+    });
+  });
+});
```
Comment (RichardJCai) [inline tests/query_interface_test.js:587]: EOF

File: tests/query_interface_test.js [hunk 2]
```diff
@@ -0,0 +1,587 @@
...
+        name: DataTypes.STRING
+      });
+      await this.queryInterface.renameTable('my_test_table', 'my_test_table_new');
+      let tableNames = await this.queryInterface.showAllTables();
+      expect(tableNames).to.contain('my_test_table_new');
+      expect(tableNames).to.not.contain('my_test_table');
+    });
+  });
+
+  // already working
```
Comment (RichardJCai) [inline tests/query_interface_test.js:71]: Is this comment necessary?

File: tests/query_interface_test.js [hunk 3]
```diff
@@ -0,0 +1,587 @@
...
+      });
+      const schemaNames = await this.queryInterface.showAllSchemas();
+      await this.queryInterface.createSchema('newSchema');
+      const newSchemaNames = await this.queryInterface.showAllSchemas();
+      expect(newSchemaNames).to.have.length(schemaNames.length + 1);
+      await this.queryInterface.dropSchema('newSchema');
+    });
+  });
+
+  // Skip only local, on CI it works
```
Comment (RichardJCai) [inline tests/query_interface_test.js:32]: ditto

File: source/patches-v6.js [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+      const query = this.queryGenerator.removeConstraintQuery(tableName, constraintName);
+      const [, queryConstraintName] = query.split('DROP CONSTRAINT');
+      const newQuery = `DROP INDEX ${queryConstraintName} CASCADE;`;
+      
+      return this.sequelize.query(newQuery, options);
+    }
+    else 
+      throw error;
+  }
+};
```
Comment (RichardJCai) [inline source/patches-v6.js:29]: EOF

File: source/patches-v6.js [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+// Drop all tables except the crdb_internal
+PostgresQueryInterface.prototype.dropSchema = async function (tableName, options) {
+  if(tableName === 'crdb_internal') return;
+  
+  await this.__dropSchema(tableName, options);
+};
+
+PostgresQueryInterface.prototype.__removeConstraint = PostgresQueryInterface.prototype.removeConstraint;
+
+// Patch drop constraint query
```
Comment (RichardJCai) [inline source/patches-v6.js:14]: Ditto

File: source/patches-v5.js [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+QueryInterface.prototype.__dropSchema = QueryInterface.prototype.dropSchema;
+
+// Drop all tables except the crdb_internal
+QueryInterface.prototype.dropSchema = async function (tableName, options) {
+  if(tableName === 'crdb_internal') return;
+  
+  await this.__dropSchema(tableName, options);
+};
+
+// Patch drop constraint query
```
Comment (RichardJCai) [inline source/patches-v5.js:12]: Ditto

File: source/index.js [hunk 7]
```diff
@@ -21,11 +21,14 @@ try {
...
 }
 
-const { Sequelize, DataTypes, QueryTypes } = require('sequelize');
+const { Sequelize, DataTypes } = require('sequelize');
 
 // Ensure Sequelize version compatibility.
 const semver = require('semver');
-const sequelizeVersion = require('sequelize/package.json').version;
+const { version, release } = require('sequelize/package.json');
+const branchVersion = release.branches ? release.branches[0] : release.branch;
```
Comment (RichardJCai) [inline source/index.js:29]: Nit: can you add a comment here explaining the logic, why we need this check for release.branch vs release.branches

**Review by RichardJCai — COMMENTED** [review]
Overall looks pretty good, mostly nits about use of comments.

**Comment by RichardJCai** [comment]
Also for all the tests we skip, please leave a note saying the reason for skipping in a format similar to the following
```
// Reason:
it.skip()
```

For all the tests that are listed as // In favor of skipping, let's do that as well and just skip them

File: tests/query_interface_test.js (hunk 2 above)
Comment (douglasselias) [reply tests/query_interface_test.js]: no, it was a self note to keep track of the tests

File: tests/query_interface_test.js [hunk 8]
```diff
@@ -0,0 +1,582 @@
...
+      });
+      const schemaNames = await this.queryInterface.showAllSchemas();
+      await this.queryInterface.createSchema('newSchema');
+      const newSchemaNames = await this.queryInterface.showAllSchemas();
+      expect(newSchemaNames).to.have.length(schemaNames.length + 1);
+      await this.queryInterface.dropSchema('newSchema');
+    });
+  });
+
+  // Reason: this test doesn't work well locally
```
Comment (RichardJCai) [inline tests/query_interface_test.js:32]: Could you leave a comment explaining why the test doesn't work locally?

**Review by RichardJCai — APPROVED** [review]
LGTM bar some small comments

File: tests/query_interface_test.js (hunk 8 above)
Comment (douglasselias) [reply tests/query_interface_test.js]: sure

File: source/index.js [hunk 9]
```diff
@@ -21,11 +21,18 @@ try {
...
 // Ensure Sequelize version compatibility.
 const semver = require('semver');
-const sequelizeVersion = require('sequelize/package.json').version;
+const { version, release } = require('sequelize/package.json');
+// In v4 and v5 package.json files, have a property 'release: { branch: 'v5' }'
+// but in v6 it has 'release: { branches: ['v6'] }'
+const branchVersion = release.branches ? release.branches[0] : release.branch;
+// When executing the tests on Github Actions the version it gets from sequelize is from the repository which has a development version '0.0.0-development'
+// in that case we fallback to a branch version
+const sequelizeVersion = semver.coerce(version === '0.0.0-development' ? branchVersion : version);
```
Comment (papb) [inline source/index.js:34]: Oh, great catch, sorry I missed this :sweat_smile:

**Pull request merged**